### ISSUE TITLE
Fail update early if dirty

### ISSUE THREAD

**Issue by pawamoy** [copier-org/copier#197]
Currently, user will be prompted with the previous answers. If the repository is dirty, update will be aborted. To avoid being prompt for nothing, the update should fail before prompting the user.

**Comment by Yajo** [comment]
I've been looking at this and it's not as easy as it seems right now:

1. To know if we're updating, first we have to make some guesses based on the results of `make_config`: https://github.com/pykong/copier/blob/a8e684736b74a4deb04cd0903d50444c8e21149d/copier/main.py#L124-L133
1. `make_config` is in charge of asking the user.
1. Doing additional calls to `make_config` clones the source more times, which slows down the thing like a lot.

Basically this is too hard to fix as the code is right now. I'll change milestone here because this would be much easier after fixing #110.

**Comment by pawamoy** [comment]
This was fixed by the refactor 🎉
